resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddressStreamProvider.java

```diff
@@ -25,12 +25,29 @@
  * /etc/resolver</a>.
  */
 @UnstableApi
-public interface DnsServerAddressStreamProvider {
+public abstract class DnsServerAddressStreamProvider {
+    // TODO(scott): how is this done on Windows? This may require a JNI call to GetNetworkParams
+    // https://msdn.microsoft.com/en-us/library/aa365968(VS.85).aspx.
+    private static final DnsServerAddressStreamProvider DEFAULT_DNS_SERVER_ADDRESS_STREAM_PROVIDER =
+            UnixResolverDnsServerAddressStreamProvider.parseSilently();
+
     /**
      * Ask this provider for the name servers to query for {@code hostname}.
      * @param hostname The hostname for which to lookup the DNS server addressed to use.
-     * @return The {@link DnsServerAddressStream} which should be used to resolve {@code hostname} or {@code null} to
-     * use the default resolvers.
+     *                 If this is the final {@link DnsServerAddressStreamProvider} to be queried then generally empty
+     *                 string or {@code '.'} correspond to the default {@link DnsServerAddressStream}.
+     * @return The {@link DnsServerAddressStream} which should be used to resolve {@code hostname}.
+     */
+    public abstract DnsServerAddressStream nameServerAddressStream(String hostname);
+
+    /**
+     * A {@link DnsServerAddressStreamProvider} which inherits the DNS servers from your local host's configuration.
+     * <p>
+     * Note that only macOS and Linux are currently supported.
+     * @return A {@link DnsServerAddressStreamProvider} which inherits the DNS servers from your local host's
+     * configuration.
      */
-    DnsServerAddressStream nameServerAddressStream(String hostname);
+    public static DnsServerAddressStreamProvider platformDefault() {
```

Comment:
I will move this method into another class and change this back to an interface. The motivation for this method creates a circular class loading issue.